Keas.Mvc/Controllers/TeamAdminController.cs

```diff
@@ -363,5 +369,172 @@ public async Task<IActionResult> RemovePpsDepartment(int id, TeamPpsDepartment p
+        {
+            var keyCount = 0;
+            var serialCount = 0;
+            var peopleCount = 0;
+            var assignmentCount = 0;
+            var rowNumber = 0;
+            StringBuilder warning = new StringBuilder();
+
+            if (file == null || file.Length == 0)
+            {
+                Message = "File not selected";
+                return RedirectToAction("Upload");
+            }
+            // Add counts
+            var team = await _context.Teams.FirstAsync(t => t.Slug == Team);
+            using (var reader = new StreamReader(file.OpenReadStream()))
+            using (var csv = new CsvReader(reader))
+            {
+                csv.Configuration.PrepareHeaderForMatch = (string header, int index) => header.ToLower();
+                var record = new KeyImport();
+                var records = csv.EnumerateRecords(record);
+                foreach (var r in records)
+                {
+                    rowNumber += 1;
+                    if (!string.IsNullOrWhiteSpace(r.Keynumber))
+                    {
+                        var key = await _context.Keys.Where(k => k.Team.Slug == Team).SingleOrDefaultAsync(k => k.Code.ToUpper() == r.Keynumber.ToUpper());
+                        if (key == null)
+                        {
+                            key = new Key();
+                            key.Code = r.Keynumber.ToUpper();
+                            key.TeamId = team.Id;
+                            key.Name = r.Description;
+
+                            TryValidateModel(key);
+                            if (ModelState.IsValid)
+                            {
+                                _context.Keys.Add(key);
+                                keyCount += 1;
+                            }
+                            else
+                            {
+                                warning.Append(String.Format("Could not save key in line {0} | ", rowNumber));
+                            }
+                        }
+
+                        if (!string.IsNullOrWhiteSpace(r.SerialNumber))
+                        {
+                            var serial = await _context.KeySerials.Where(s => s.KeyId == key.Id).SingleOrDefaultAsync(s => s.Number.ToUpper() == r.SerialNumber.ToUpper());
+                            if (serial == null)
+                            {
+                                serial = new KeySerial();
+                                serial.Number = r.SerialNumber;
+                                serial.Name = r.SerialNumber;
+                                serial.Key = key;
+                                serial.Status = "Active";
+                                if(r.Status == "L" || r.Status == "Lost")
+                                {
+                                    serial.Status = "Lost";
+                                }
+                                if(r.Status == "B" || r.Status == "Destroyed")
```

Comment:
@jcubbage We shouldn't be translating "their" codes. They should replace "B" in their file.